cmd/skunk/adapter/in/networkAdapter/networkAdapter.go

```diff
@@ -0,0 +1,250 @@
+package networkAdapter
+
+import (
+	"encoding/json"
+	"fmt"
+	"sync"
+	"time"
+
+	cretztor "github.com/cretz/bine/tor"
+
+	"github.com/scherzma/Skunk/cmd/skunk/adapter/in/peer"
+	"github.com/scherzma/Skunk/cmd/skunk/adapter/in/tor"
+	"github.com/scherzma/Skunk/cmd/skunk/application/port/network"
+	"github.com/scherzma/Skunk/cmd/skunk/util"
+)
+
+// Constants for configuration of Tor
+const (
+	SocksPort            = "9055"
+	LocalPort            = "1111"
+	RemotePort           = "2222"
+	ReusePrivateKey      = true // reuse private key for constant onion address
+	DeleteDataDirOnClose = false
+	UseEmbedded          = true // use embedded tor process
+)
+
+var (
+	networkAdapter *NetworkAdapter
+	once           sync.Once
+)
+
+// NetworkAdapter connects the main logic to the tor peer network
+type NetworkAdapter struct {
+	subscriber network.NetworkObserver // subscriber observing network messages
+	peer       *peer.Peer
+	tor        *tor.Tor
+}
+
+func NewAdapter() *NetworkAdapter {
+	// singleton
+	once.Do(func() {
+		networkAdapter = &NetworkAdapter{}
+	})
+
+	return networkAdapter
+}
+
+func (n *NetworkAdapter) SubscribeToNetwork(observer network.NetworkObserver) error {
+	if n.subscriber == observer {
+		return fmt.Errorf("network adapter is already connected to observer: %v", observer)
+	}
+
+	if n.peer != nil || n.tor != nil {
+		return fmt.Errorf("network services are already running")
+	}
+
+	torInstance, onionService, err := startTor()
+	if err != nil {
+		return err
+	}
+
+	peerInstance, err := startPeer(onionService)
+	if err != nil {
+		return err
+	}
+
+	// begin asynchronously reading network messages.
+	go n.readNetworkMessages()
+
+	n.subscriber = observer
+	n.peer = peerInstance
+	n.tor = torInstance
+
+	// notify the subscriber that the network is now online and send the onion address
+	message := network.Message{
+		Id:              util.UUID(),
+		Timestamp:       util.CurrentTimeMillis(),
+		Content:         n.peer.Address,
```

Comment:
We can send it as a normal string, but so far everything in a messageContent is JSON. Maybe we should also make this JSON.

Reply:
Done in: [ea02765](https://github.com/scherzma/Skunk/pull/33/commits/ea027653036eff1b5fd8c22e4d9656ae0a8c2187)